pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -211,7 +211,7 @@ public static String getRemoteCluster(String remoteCursor) {
     }
 
     public static String getReplicatorName(String replicatorPrefix, String cluster) {
-        return String.format("%s.%s", replicatorPrefix, cluster);
+        return (replicatorPrefix + "." + cluster).intern();
```

Comment:
Couldn't the string be interned also after `String.format()` ?

Reply:
actually I ran a test where I saw that `String.format("%s%s",x,y).intern()` allocates some more number of string objects than `(x+"."+y).intern()` when I checked with `jmap -histo <pid> | grep String`. However, `intern()` still reuses objects in both cases but somehow, `String.format()` adds some more instances compare to append.